contracts/VeJoeStaking.sol

```diff
@@ -0,0 +1,267 @@
+        baseGenerationRate = _baseGenerationRate;
+        boostedGenerationRate = _boostedGenerationRate;
+        boostedThreshold = _boostedThreshold;
+        boostedDuration = _boostedDuration;
+    }
+
+    /// @notice Set maxCap
+    /// @param _maxCap the new maxCap
+    function setMaxCap(uint256 _maxCap) external onlyOwner {
+        require(_maxCap > 0, "VeJoeStaking: expected new _maxCap to be greater than 0");
+        maxCap = _maxCap;
+    }
+
+    /// @notice Set baseGenerationRate
+    /// @param _baseGenerationRate the new baseGenerationRate
+    function setBaseGenerationRate(uint256 _baseGenerationRate) external onlyOwner {
+        require(
+            _baseGenerationRate < boostedGenerationRate,
+            "VeJoeStaking: expected new _baseGenerationRate to be less than boostedGenerationRate"
+        );
+        baseGenerationRate = _baseGenerationRate;
+    }
+
+    /// @notice Set boostedGenerationRate
+    /// @param _boostedGenerationRate the new boostedGenerationRate
+    function setBoostedGenerationRate(uint256 _boostedGenerationRate) external onlyOwner {
+        require(
+            _boostedGenerationRate > baseGenerationRate,
+            "VeJoeStaking: expected new _boostedGenerationRate to be greater than baseGenerationRate"
+        );
+        boostedGenerationRate = _boostedGenerationRate;
+    }
+
+    /// @notice Set boostedThreshold
+    /// @param _boostedThreshold the new boostedThreshold
+    function setBoostedThreshold(uint256 _boostedThreshold) external onlyOwner {
+        require(
+            _boostedThreshold <= 100,
+            "VeJoeStaking: expected new _boostedThreshold to be less than or equal to 100"
+        );
+        boostedThreshold = _boostedThreshold;
+    }
+
+    /// @notice Set boostedDuration
+    /// @param _boostedDuration the new boostedDuration
+    function setBoostedDuration(uint256 _boostedDuration) external onlyOwner {
+        boostedDuration = _boostedDuration;
+    }
+
+    /// @notice Deposits JOE to start staking for veJOE. Note that any pending veJOE
+    /// will also be claimed in the process.
+    /// @param _amount the amount of JOE to deposit
+    function deposit(uint256 _amount) external {
+        require(_amount > 0, "VeJoeStaking: expected deposit amount to be greater than zero");
+
+        if (getUserHasNonZeroBalance(msg.sender)) {
+            // If user already has staked JOE, we first send them any pending veJOE
+            _claim();
+
+            userInfos[msg.sender].balance += _amount;
+
+            // User is eligible for boosted benefits if:
+            // 1. They are not already currently receiving boosted benefits
+            // 2. Their staked JOE is at least `boostedThreshold / 100 * totalStakedJoe`
+            if (userInfos[msg.sender].boostEndTimestamp == 0) {
+                uint256 totalStakedJoe = joe.balanceOf(address(this));
+                if ((userInfos[msg.sender].balance * 100) / boostedThreshold >= totalStakedJoe) {
```

Comment:
Do I understand this correctly. The minimum requirement to receive a boost it to hold at least 1% of the entire pool? This is assuming `boostedThreshold` is 1 (the minimum value).

Reply:
It turns out I misunderstood this when I first wrote this two weeks ago and have updated the logic now. The correct behavior here is that the user will receive boosted benefits if they stake 5% of their current staked JOE balance and they are not already receiving boosted benefits.

For example, if the user currently has `100 JOE` staked and aren't receiving boosted benefits, they need to stake `5 JOE` to enter a boost period